### PR TITLE
Add Alerta Output

### PR DESCRIPTION
**Pull request by LivingInSyn**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area config

/area outputs

> /area tests

This PR allows post events to Alerta (https://alerta.io/) 

**Special notes for your reviewer**:

Alerta API reference: https://docs.alerta.io/api/reference.html#alerts

### CODE REVIEW

File: outputs/alerta.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	return ap
+}
+
+func (c *Client) AlertaPost(falcopayload types.FalcoPayload) {
+	c.Stats.Alertmanager.Add(Total, 1)
+	c.AddHeader(AuthorizationHeaderKey, fmt.Sprintf("Key %s", c.Config.Alerta.AuthKey))
+
+	err := c.Post(newAlertaPayload(falcopayload, c.Config))
+	if err != nil {
+		go c.CountMetric(Outputs, 1, []string{"output:alertmanager", "status:error"})
```
Comment (Issif) [inline outputs/alerta.go:72]:
```suggestion
		go c.CountMetric(Outputs, 1, []string{"output:alerta", "status:error"})
```

File: outputs/alerta.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	// set rawdata to the json encoded falcopayload
+	rdb, err := json.Marshal(falcopayload)
+	if err == nil {
+		ap.RawData = string(rdb)
+	}
+	return ap
+}
+
+func (c *Client) AlertaPost(falcopayload types.FalcoPayload) {
+	c.Stats.Alertmanager.Add(Total, 1)
```
Comment (Issif) [inline outputs/alerta.go:67]:
```suggestion
	c.Stats.Alerta.Add(Total, 1)
```

File: outputs/alerta.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	c.AddHeader(AuthorizationHeaderKey, fmt.Sprintf("Key %s", c.Config.Alerta.AuthKey))
+
+	err := c.Post(newAlertaPayload(falcopayload, c.Config))
+	if err != nil {
+		go c.CountMetric(Outputs, 1, []string{"output:alertmanager", "status:error"})
+		c.Stats.Alertmanager.Add(Error, 1)
```
Comment (Issif) [inline outputs/alerta.go:73]:
```suggestion
		c.Stats.Alerta.Add(Error, 1)
```

File: outputs/alerta.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+		c.PromStats.Outputs.With(map[string]string{"destination": "alerta", "status": Error}).Inc()
+		log.Printf("[ERROR] : Alerta - %v\n", err)
+		return
+	}
+
+	go c.CountMetric(Outputs, 1, []string{"output:alerta", "status:ok"})
+	c.Stats.Alertmanager.Add(OK, 1)
```
Comment (Issif) [inline outputs/alerta.go:80]:
```suggestion
	c.Stats.Alerta.Add(OK, 1)
```

**Review by Issif — CHANGES_REQUESTED** [review]

**Comment by LivingInSyn** [comment]
@Issif fixed alertmanager-> alerta issues, that's what I get for not waiting until the next morning to submit the PR, sorry about that

File: handlers.go [hunk 5]
```diff
@@ -125,6 +125,10 @@ func forwardEvent(falcopayload types.FalcoPayload) {
 		go cliqClient.CliqPost(falcopayload)
 	}
 
+	if config.Alerta.Address != "" && (falcopayload.Priority >= types.Priority(config.Cliq.MinimumPriority) || falcopayload.Rule == testRule) {
```
Comment (Issif) [inline handlers.go:128]:
```suggestion
	if config.Alerta.Address != "" && (falcopayload.Priority >= types.Priority(config.Alerta.MinimumPriority) || falcopayload.Rule == testRule) {
```

**Review by Issif — CHANGES_REQUESTED** [review]

**Review by Issif — COMMENTED** [review]
Seems OK to me, can you squash your commits please, I'll approve then

**Comment by LivingInSyn** [comment]
I'm closing this PR due to discovered issues in Alerta and inconsistent rules in their API between versions and in their docs. I no longer think Alerta is stable enough to not break

**Pull request closed without merging**